pkgs/by-name/sq/sqlite-vss/package.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, cmake
+, faiss
+, fetchFromGitHub
+, gomp
+, llvmPackages
+, nlohmann_json
+, sqlite
+, stdenv
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sqlite-vss";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "asg017";
+    repo = "sqlite-vss";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-cb9UlSUAZp8B5NpNDBvJ2+ung98gjVKLxrM2Ek9fOcs=";
+  };
+
+  patches = [ ./use-nixpkgs-libs.patch ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ nlohmann_json faiss sqlite ]
+    ++ lib.optional stdenv.isLinux gomp
+    ++ lib.optional stdenv.isDarwin llvmPackages.openmp;
+
+  SQLITE_VSS_CMAKE_VERSION = finalAttrs.version;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib
+
+    cp libsqlite_vector0${stdenv.hostPlatform.extensions.staticLibrary} $out/lib
+    cp libsqlite_vss0${stdenv.hostPlatform.extensions.staticLibrary} $out/lib
+    cp vector0${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib
+    cp vss0${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib
```

This can be more easily accomplished in a single statement

```suggestion
    install -Dm444 -t "$out/lib" \
      "libsqlite_vector0${stdenv.hostPlatform.extensions.staticLibrary}" \
      "libsqlite_vss0${stdenv.hostPlatform.extensions.staticLibrary}" \
      "vector0${stdenv.hostPlatform.extensions.sharedLibrary}" \
      "vss0${stdenv.hostPlatform.extensions.sharedLibrary}"
```